tests/test_predict_kge_literals.py

```diff
@@ -0,0 +1,233 @@
+import os
+import pytest
+import shutil
+import numpy as np
+import pandas as pd
+from dicee.config import Namespace
+from dicee.executer import Execute
+from dicee.knowledge_graph_embeddings import KGE
+
+class TestPredictLitRegression:
+    """Regression tests for literal prediction using interactive KGE model Family dataset."""
+    
+    @pytest.fixture(scope="function", autouse=True)
+    def generate_literal_files(self):
+        """
+        Generate training and testing files for literal triples.
+
+        Args:
+            file_path (str): The directory where the files will be saved.
+        """
+
+        literal_triples =     [
+            ('F5F62', 'Age', 18), ('F1F3', 'Height', 164),
+            ('F6M99', 'Height', 138), ('F6F97', 'Age', 17),
+            ('F6M92', 'Weight', 80), ('F3F41', 'Age', 59),
+            ('F9F141', 'Age', 53), ('F2F33', 'Age', 39),
+            ('F2M35', 'Height', 138), ('F3F46', 'Height', 140),
+            ('F9M155', 'Weight', 77), ('F6F70', 'Height', 152),
+            ('F7F121', 'Age', 91), ('F2M23', 'Weight', 75),
+            ('F2F24', 'Weight', 57), ('F9M162', 'Age', 42),
+            ('F1F7', 'Weight', 62), ('F5M60', 'Height', 151),
+            ('F10F181', 'Weight', 60), ('F3M47', 'Age', 40),
+            ('F2M16', 'Height', 163), ('F6F84', 'Height', 149),
+            ('F10F198', 'Age', 36), ('F7M125', 'Weight', 40),
+            ('F10M173', 'Height', 177)
+        ]
+        #train test splits
+        train_triples = literal_triples[:20]
+        test_triples = literal_triples[20:]
+
+        # Output directory
+        output_dir =  "KGs/Family/literals"
+        os.makedirs(output_dir, exist_ok=True)
+
+        train_df = pd.DataFrame(train_triples)
+        test_df = pd.DataFrame(test_triples)
+        train_df.to_csv( os.path.join(output_dir, "train.txt"), header=False, sep ="\t",index=False)
+        test_df.to_csv( os.path.join(output_dir, "test.txt"), header=False,sep ="\t", index=False)
+
+
+    @pytest.fixture(scope="class")
+    def family_model(self):
+        """Setup Keci model trained on Family dataset."""
+
+        # Create training triples for the Family dataset.
+        self.create_train_triples(path="KGs/Family")
```

Comment:
You may want to reorder the function definition orders. create_train_triples should be defined before it is being used